scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala

```diff
@@ -20,7 +20,7 @@ trait GrantHandler {
    */
   def clientCredentialRequired = true
 
-  def handleRequest[U](request: AuthorizationRequest, authorizationHandler: AuthorizationHandler[U])(implicit ctx: ExecutionContext): Future[GrantHandlerResult[U]]
+  def handleRequest[U](validatedClientId: Option[String], request: AuthorizationRequest, authorizationHandler: AuthorizationHandler[U])(implicit ctx: ExecutionContext): Future[GrantHandlerResult[U]]
```

Comment:
How about the type changes to `Option[ClientCredential]`?

`AuthorizationRequest` has `clientCredential` instance and `GrantHandler#handleRequest` method receive validated `clientId` using `AuthorizationRequest` instance and `AuthorizationRequest` instance itself. `GrantHandler#handleRequest` method requires duplicated variable. I want to avoid it.

I have an idea that solve the problem.

- Change `AuthorizationRequest#clientCredential` instance to `def parseClientCredential: Option[Either[InvalidClient, ClientCredential]` method
- Change `GrantHandler#handleRequest` first of arguments type to `maybeClientCredential: Option[ClientCredential]`

What do you think?